maven/flatpak-maven-generator

```diff
@@ -0,0 +1,222 @@
+                           "dest": f'maven-local/{groupId}/{artifactId}/{version}'
+                        })
+
+                modules.append({
+                   "type": "file",
+                   "url": url,
+                   "sha256": getFileHash(tmp_file),
+                   "dest": f"maven-local/{groupId}/{artifactId}/{version}"
+                })
+
+                return True
+    except urllib.error.HTTPError as e:
+        print("{colors.WARNING}Unable to find get module metadata, even though it was specified{colors.ENDC}")
+        return False
+
+def parsePomTree(repos: list[str], groupId: str, artifactId: str, version: str) -> bool:
+    for module in addedModules:
+        if module["groupId"] == groupId and module["artifactId"] == artifactId and module["version"] == version:
+            return True
+
+    for repo in repos:
+        url = assembleUri(repo, groupId, artifactId, version, "pom")
+
+        try:
+            print(f"Looking for {artifactId}:{version} at {url}")
+            with urllib.request.urlopen(url) as response:
+                with tempfile.NamedTemporaryFile(delete=True) as tmp_file:
+                    addModule(groupId, artifactId, version)
+
+                    shutil.copyfileobj(response, tmp_file)
+                    tmp_file.seek(0)
+                    file_content = tmp_file.read().decode('utf-8')
+                    parsed_file = ET.fromstring(file_content)
+
+                    if (binaryType := getPackagingType(parsed_file)) is not None:
+                        downloadAndAdd(repo, groupId, artifactId, version, binaryType)
+
+                    if("do_not_remove: published-with-gradle-metadata" in file_content):
+                        # This module has extended Gradle metadata, download that (and its dependencies)
+                        parseGradleMetadata(repo, groupId, artifactId, version)
+
+                    deps = parsePomDeps(parsed_file)
+                    for dep in deps:
+                        parsePomTree(repos, dep["groupId"], dep["artifactId"], dep["version"])
+
+                    groupId = groupId.replace(".", "/")
+                    modules.append({
+                        "type": "file",
+                        "url": url,
+                        "sha256": getFileHash(tmp_file),
+                        "dest": f"maven-local/{groupId}/{artifactId}/{version}"
+                    })
+
+                    return True
+
+        except urllib.error.HTTPError as e:
+            pass
+
+    print(f"{colors.WARNING}{artifactId}:{version} not found in any of the specified repos{colors.ENDC}")
+    return False
+
+for package in opts.packages:
+    package_parts = package.split(":")
+    if len(package_parts) != 3:
+        print("{colors.FAIL}Package names must be in the format groupId:artifactId:version{colors.ENDC}")
+        sys.exit(1)
+
+    groupId = package_parts[0]
+    artifactId = package_parts[1]
+    version = package_parts[2]
```

Comment:
These variable names are used in some functions. Better rename them in the module scope to avoid redefining, or just wrap all the module-scope code into some `main()` function.